### PR TITLE
Search Spots by bounds

### PR DESCRIPTION
**Pull request by TNCodeMonkey**
I've added some functionality to search Spots by bounds, found it really useful in an application I was working on.  I am very new to open source work, so please let me know if I've missed anything or done something wrong.  Thanks!

### CODE REVIEW

File: lib/google_places/client.rb [hunk 1]
```diff
@@ -148,7 +148,40 @@ def spot(place_id, options = {})
...
+    # @option options [String,Array<String>] :exclude ([])
+    #   A String or an Array of <b>types</b> to exclude from results
+    #
+    # @option options [Hash] :retry_options ({})
+    #   A Hash containing parameters for search retries
+    # @option options [Object] :retry_options[:status] ([])
+    # @option options [Integer] :retry_options[:max] (0) the maximum retries
+    # @option options [Integer] :retry_options[:delay] (5) the delay between each retry in seconds
+    #
+    # @see http://spreadsheets.google.com/pub?key=p9pdwsai2hDMsLkXsoM05KQ&gid=1 List of supported languages
```
Comment (marceldegraaf) [inline lib/google_places/client.rb]: When opening this link I get an error: "We're sorry, this document is not published". Can you fix the link to a working version?

File: lib/google_places/rectangle.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    def initialize(lat1, lng1, lat2, lng2)
+      @lat1 = ("%.8f"%lat1)
+      @lng1 = ("%.8f"%lng1)
+      @lat2 = ("%.8f"%lat2)
+      @lng2 = ("%.8f"%lng2)
+    end
+
+    def format
+      #SW point of rectangle
+      a1 = [@lat1, @lat2].join (',')
```
Comment (marceldegraaf) [inline lib/google_places/rectangle.rb]: I would add spaces inside the Array (e.g. `[ @lat1, @lat2 ]`) and remove the space in `join (',')`

File: lib/google_places/spot.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 require 'google_places/review'
-
+require 'google_places'
```
Comment (marceldegraaf) [inline lib/google_places/spot.rb]: Why did you add this `require`?

File: lib/google_places/spot.rb [hunk 4]
```diff
@@ -82,7 +82,67 @@ def self.list(lat, lng, api_key, options = {})
...
+      # @option options [Hash] :retry_options ({})
+      #   A Hash containing parameters for search retries
+      # @option options [Object] :retry_options[:status] ([])
+      # @option options [Integer] :retry_options[:max] (0) the maximum retries
+      # @option options [Integer] :retry_options[:delay] (5) the delay between each retry in seconds
+      #
+      # @see http://spreadsheets.google.com/pub?key=p9pdwsai2hDMsLkXsoM05KQ&gid=1 List of supported languages
+      # @see https://developers.google.com/maps/documentation/places/supported_types List of supported types
+    def self.list_by_bounds(bounds, api_key, options = {})
+      multipage_request = !!options.delete(:multipage)
```
Comment (marceldegraaf) [inline lib/google_places/spot.rb]: Can you refactor `!!options...` to something a bit more verbose? It's hard to see what's going on here.

File: lib/google_places/spot.rb [hunk 5]
```diff
@@ -82,7 +82,67 @@ def self.list(lat, lng, api_key, options = {})
...
+      rankby = options.delete(:rankby)
+      rect = Rectangle.new(bounds[:sw][:lat], bounds[:sw][:lng],bounds[:ne][:lat], bounds[:ne][:lng])
```
Comment (marceldegraaf) [inline lib/google_places/spot.rb]: Missing a space between `,bounds[:ne][:lat]...`

File: lib/google_places/rectangle.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
+module GooglePlaces
+  class Rectangle
+    def initialize(lat1, lng1, lat2, lng2)
```
Comment (marceldegraaf) [inline lib/google_places/rectangle.rb]: Isn't it more understandable to use something like `north_west`, `north_east`, `south_east`, and `south_west` here, to indicate the outer points of the rectangle? Latitude and longitude may be a bit too abstract?

**Comment by marceldegraaf** [comment]
Thanks @TNCodeMonkey, looks good, aside from a few formatting issues. If you can take a look at those, I'd be happy to merge this PR in and push out a new gem version.

File: lib/google_places/client.rb (hunk 1 above)
Comment (TNCodeMonkey) [inline lib/google_places/client.rb]: Thank you for the feedback.  I will knock those out very soon!  Your lib
was awesome for my application, just glad I could give back.

--Dustin

Comment (marceldegraaf) [inline lib/google_places/client.rb]: Awesome, looking forward to it! :smile:

File: lib/google_places/spot.rb (hunk 3 above)
Comment (TNCodeMonkey) [inline lib/google_places/spot.rb]: We can chalk this up to being a Ruby noobie :P .  Pulling this out for next commit

Comment (marceldegraaf) [inline lib/google_places/spot.rb]: No problem! :+1:

File: lib/google_places/rectangle.rb (hunk 6 above)
Comment (TNCodeMonkey) [inline lib/google_places/rectangle.rb]: I wanted to leave the option option to create the rectangle from any start+end point combination -- NE + SW, SE + NW, NW + SE, and so on.  I was afraid that specifying SE/NW would infer to the user that they could only define bounds by that given pair.  Do you think start_lat, start_lng, end_lat, end_lng would gain us some clarity?

Comment (marceldegraaf) [inline lib/google_places/rectangle.rb]: Alright, if you can make sure to document (in a comment) sufficiently what these variables do and mean, it's fine with me :smile:

**Comment by TNCodeMonkey** [comment]
Ok @marceldegraaf , I think I've updated everything that needed attention.  Also, I went ahead and added test coverage for rectangle and Client.spots_by_bounds.  Please let me know if you need anything else taken care of with these changes.  Thanks!

**Comment by marceldegraaf** [comment]
Awesome, thanks! Version 0.30.0 of the gem was just released, and it contains this addition. Nice work @TNCodeMonkey :smile:

**Comment by TNCodeMonkey** [comment]
Thank you for your critique, actually made my first OSS contribution very painless :)  Look forward to adding more in the future!!

**Comment by marceldegraaf** [comment]
No problem!
